src/utils/Fetch.ts

```diff
@@ -67,11 +57,26 @@ const fetch = <T>(url: string): Promise<T> => {
     request.on('socket', socket => {
       socket.on('timeout', () => {
         request.abort();
-        reject(`Socket timeout while connecting to '${url}'`);
+        reject(`Socket timeout while connecting to '${options.host}${options.path}'`);
       });
       socket.setTimeout(SOCKET_TIMEOUT);
     });
-  });
-};
 
-export { fetch };
+    request.end();
+  })
+}
+
+/**
+ * Fetch JSON data from a remote HTTP endpoint with the provided headers and de-serialize to the provided type.
+ * There are no guarantees the response will conform to the contract defined by T.
+ * It is up to the consumer to ensure the provided T captures all possible response types
+ * from the provided endpoint.
+ *
+ * @param options - HTTP request options
+ */
+const fetch = async <T>(options: RequestOptions): Promise<T> => {
```

Comment:
Since we are already adjusting the callers of `fetch` to send options, can we also rename this method to `fetchJSON` for some more consistency & transparency?